brian2/core/network.py

```diff
@@ -526,7 +556,8 @@ def store(self, name='default', filename=None):
         versions. If you are interested in storing the state of a network for
         documentation or analysis purposes use `Network.get_states` instead.
         '''
-        clocks = [obj.clock for obj in self.objects]
+        all_objects = self.get_all_objects()
+        clocks = [obj.clock for obj in all_objects]
```

Comment:
Again, already a shortcoming of the previous code, but this should be rather a set (`{...}`) instead of a list, we don't need to store the same `Clock` repeatedly.